types/wlr_output.c

```diff
@@ -51,6 +51,25 @@ static void wl_output_send_to_resource(struct wl_resource *resource) {
 	}
 }
 
+static void wlr_output_send_current_mode_to_resource(
+		struct wl_resource *resource) {
+	struct wlr_output *output = wl_resource_get_user_data(resource);
+	assert(output);
+	const uint32_t version = wl_resource_get_version(resource);
+	if (version < WL_OUTPUT_MODE_SINCE_VERSION) {
+		return;
+	}
+	if (output->current_mode != NULL) {
+		struct wlr_output_mode *mode = output->current_mode;
+		wl_output_send_mode(resource, mode->flags | WL_OUTPUT_MODE_CURRENT,
+			mode->width, mode->height, mode->refresh);
```

Comment:
Do you have to destroy the old mode we sent? Maybe there's a way to modify it?

Reply:
I think `output.current_mode` is just a pointer to an element of `output->modes`, so it doesn't matter.

Reply:
(Ah sorry, misunderstood, was thinking about memory, not protocol)

https://wayland.freedesktop.org/docs/html/apa.html#protocol-spec-wl_output-event-mode

>The event is sent when binding to the output object and there will always be one mode, the current mode. The event is sent again if an output changes mode, for the mode that is now current. In other words, the current mode is always the last mode that was received with the current flag set.